pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -262,8 +296,146 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 	return preds
 }
 
-func (s *cyclonedxParser) getPackageElement(elementID string) []*model.PkgInputSpec {
-	if packNode, ok := s.packagePackages[string(elementID)]; ok {
+func (c *cyclonedxParser) getVulnerabilities(ctx context.Context, cdxBom *cdx.BOM) *vulnData {
+	if cdxBom.Vulnerabilities == nil {
+		return nil
+	}
+
+	var vInfo vulnData
+	var status model.VexStatus
+	var justification model.VexJustification
+	var publishedTime time.Time
+	layout := "2006-01-02T15:04:05.000Z"
+	for _, vulnerability := range *c.cdxBom.Vulnerabilities {
+		vuln, err := asmhelpers.CreateVulnInput(vulnerability.ID)
+		if err != nil {
+			return nil
+		}
+
+		if vexStatus, ok := vexStatusMap[vulnerability.Analysis.State]; ok {
+			status = vexStatus
```

Comment:
will this always be ok?

Reply:
No but do we want to fallback to a default value in that case?

Reply:
same here move to parse function and return an error if not found